lib/intl.mjs

```diff
@@ -41,21 +43,44 @@ const descriptor = (value) => {
 const IntlDateTimeFormat = globalThis.Intl.DateTimeFormat;
 const ObjectAssign = Object.assign;
 
-export function DateTimeFormat(locale = IntlDateTimeFormat().resolvedOptions().locale, options = {}) {
+// Construction of  built-in Intl.DateTimeFormat objects is sloooooow,
+// so we'll only create those instances when we need them.
+// See https://bugs.chromium.org/p/v8/issues/detail?id=6528
+function getPropLazy(obj, prop) {
+  let val = obj[prop];
+  if (prop === TZ_RESOLVED) {
+    // for TimeZone, the initializer is a string, not a function
+    if (typeof val === 'string') {
+      val = new TimeZone(val);
+      obj[prop] = val;
+    }
+  } else if (typeof val === 'function') {
+    val = new IntlDateTimeFormat(obj[LOCALE], val(obj[OPTIONS]));
+    obj[prop] = val;
+  }
+  return val;
+}
+
+export function DateTimeFormat(locale = undefined, options = {}) {
   if (!(this instanceof DateTimeFormat)) return new DateTimeFormat(locale, options);
+  const original = new IntlDateTimeFormat(locale, options);
+  const ro = original.resolvedOptions();
+  options = typeof options === 'undefined' ? {} : ObjectAssign({}, options);
```

Comment:
Yep, I made exactly this change in my latest checkin.  I also realized that shallow-cloning of `options` was not enough, because if `options` property values or `locale` were not primitives, then they could also be mutated by the caller before lazy-init happens. Take a look at my last checkin and let me know if the hacky JSON-based deep clone is sufficient.

Reply:
I also added tests to verify that mutating constructor inputs after the constructor returns won't break anything.

Still not sure that using a JSON round-trip is the right way to deep-clone these inputs.